databases/functions/userFollowUser/user-follow-add.js

```diff
@@ -21,20 +21,36 @@ module.exports = async (
   let follows_array = [];
   await Promise.all(
     followedUsers.map(async (val) => {
-      //follow sign user
-      follows_array.push({
-        follow_action_id: uuidv4(),
-        user_id_follower: userId,
-        user_id_followed: val
-      });
-
       const userFollowed = await UsersFunction.findUserById(User, val);
       if (userFollowed.is_anonymous === false) {
+        follows_array.push({
+          follow_action_id: uuidv4(),
+          user_id_follower: userId,
+          user_id_followed: val
+        });
+
         const anonymousUser = await UsersFunction.findAnonymousUserId(User, val);
+        let isAlreadyFollowAnonUser = await userFollowUserModel.findOne({
+          where: {
+            user_id_follower: userId,
+            user_id_followed: anonymousUser?.user_id,
+            is_anonymous: true
+          }
+        });
+
+        if (!isAlreadyFollowAnonUser) {
+          follows_array.push({
+            follow_action_id: uuidv4(),
+            user_id_follower: userId,
+            user_id_followed: anonymousUser?.user_id,
+            is_anonymous: true
+          });
+        }
+      } else if (userFollowed.is_anonymous === true) {
         follows_array.push({
           follow_action_id: uuidv4(),
           user_id_follower: userId,
-          user_id_followed: anonymousUser?.user_id,
+          user_id_followed: val,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for following users, both anonymous and non-anonymous, seems to be correct. However, there is a potential performance issue here. For each user in the `followedUsers` array, you are making two database calls: one to find the user by ID and another to find the anonymous user ID (if the user is not anonymous). This could lead to a significant number of database calls if the `followedUsers` array is large.

A more efficient approach would be to fetch all the necessary data in fewer database calls before entering the loop. You can fetch all users and their corresponding anonymous users at once, then use this data inside your loop. This way, you only make two database calls regardless of the size of the `followedUsers` array.

Here's an example of how you might do this:

```diff
-  await Promise.all(
-    followedUsers.map(async (val) => {
-      const userFollowed = await UsersFunction.findUserById(User, val);
-      ...
-      const anonymousUser = await UsersFunction.findAnonymousUserId(User, val);
-      ...
-    })
-  );

+  // Fetch all users and their corresponding anonymous users at once
+  const users = await UsersFunction.findUsersByIds(User, followedUsers);
+  const anonymousUsers = await UsersFunction.findAnonymousUserIds(User, followedUsers);

+  await Promise.all(
+    followedUsers.map(async (val, index) => {
+      const userFollowed = users[index];
+      ...
+      const anonymousUser = anonymousUsers[index];
+      ...
+    })
+  );
```

In this code, `findUsersByIds` and `findAnonymousUserIds` are hypothetical functions that you would need to implement. They should take an array of user IDs and return an array of user objects or anonymous user IDs, respectively.


<!-- This is an auto-generated comment by OSS CodeRabbit -->